languages/emf-metamodel/src/main/java/de/jplag/emf/util/ModelingElementIdentifierManager.java

```diff
@@ -0,0 +1,42 @@
+package de.jplag.emf.util;
+
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+
+/**
+ * This class provides type-unique identifiers for EObjects.
+ */
+public class ModelingElementIdentifierManager {
+
+    private final Map<EClass, Set<EObject>> elementToIdentifer;
+
+    /**
+     * Creates the identifier manager. Identifiers are only unique if managed by the same instance.
+     */
+    public ModelingElementIdentifierManager() {
+        elementToIdentifer = new HashMap<>();
+    }
+
+    /**
+     * Returns the type-unique identifier for any EMF modeling element.
+     * @param element is the modeling element for which the identifier is requested.
+     * @return the identifier, that is unique for all elements of the same EClass.
+     */
+    public int getIdentifier(EObject element) {
+        Set<EObject> elements = elementToIdentifer.computeIfAbsent(element.eClass(), key -> new LinkedHashSet<>());
+        int index = 0;
+        for (EObject containedElement : elements) {
```

Comment:
This should be a normal for loop with index as variable

Reply:
Note that this is a set, thus there is no indexed access.

Reply:
If you cannot access via index,why do you need to return the "index" where the element parameter is?

Reply:
That is the whole purpose of the class. What data structure would you use if you need uniqueness (set) and natural order?